py3status/modules/emerge_status.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+
+"""
+Emerge_status provides a short information on the current
+emerge status if there is an emerge process currently running
+
+Configuration parameters:
+    cache_timeout: how often we refresh this module in second (default 1)
+    emerge_log_file: in case it is placed somewhere else than the default
+        (default '/var/log/emerge.log')
+    format: Display format to user (default '{current} / {total}')
+    hide_if_zero: Don't show in bar if there is no emerge running (default True)
+
+    More format options:
+        Ex: '{current} / {total}'
+        Result: "3 / 14"
+        Ex: '{current} / {total} = {'
+        Result: "3 / 14 = py3status"
+        Ex: '{category} - {pkg}'
+        Result: "x11-misc - py3status"
+
+@author: AnwariasEu
+"""
+
+import os
+import re
+
+
+class Py3status:
+    """
+    """
+    cache_timeout = 1
+    emerge_log_file = '/var/log/emerge.log'
+    format = u'{current} / {total}'
+    hide_if_zero = True
+
+    def _emergeRunning(self):
+        """
+        Use pgrep to check if emerge is running.
+
+        Returns true if at least one instance of emerge is running.
+        """
+        tmp = os.popen("pgrep emerge").read()
+        pcount = tmp.count("\n")
+        if pcount > 0:
+            return True
+        else:
+            return False
+
+    def _getProgress(self):
+        """
+        Get current progress of emerge.
+
+        returns a dict containing current and total value.
+        """
+        ret = {}
+        ret['current'] = None
+        ret['total'] = None
+        ret['err'] = False
+        ret['err_text'] = ""
+        ret['category'] = None
+        ret['pkg'] = None
```

Comment:
See if we can simplify things enough to start with just `ret = {}`.

Reply:
I guess this change only makes sense if the undocumented modifiers like `\?if`  are used?

Reply:
This is to kill lines. The format can use `[{example}]` to make non-existent  placeholders and `None` placeholders go away. It looks like some values would be filled one way or other in the loop.

Or you could leave `ret = {}` and move the others down to `*** terminating line`. Use `''` instead of `None` (to make it look nice).